tests/suites/host_test.function

```diff
@@ -368,6 +368,118 @@ static int run_test_snprintf( void )
             test_snprintf( 5, "123",         3 ) != 0 );
 }
 
+/** \brief Write the description of the test case to the outcome CSV file.
+ *
+ * \param outcome_file  The file to write to.
+ *                      If this is \c NULL, this function does nothing.
+ * \param argv0         The test suite name.
+ * \param test_case     The test case description.
+ */
+static void write_outcome_entry( FILE *outcome_file,
```

Comment:
Ah, yes, that's a good point. It should. But if it does, we won't be able to collect test outcomes in builds without `MBEDTLS_FS_IO`. I chose this approach because it seemed to be simpler, but I hadn't thought of this. So I should switch to collecting the output in scripts, but that's a more invasive change because it requires storing the output somewhere or piping it.

Reply:
To avoid scope creep, I think I'm going to stick with this approach for now, and document the limitation that the outcomes of tests without `MBEDTLS_FS_IO` are not recorded. I'd rather not change how `make test` works in this PR.

Reply:
Actually, scratch this. In the test suites, `MBEDTLS_FS_IO` is not required in the **host** test automation: we assume that stdio is available. So the code is fine as is.

It's a known limitation that this PR doesn't record the outcome of on-target tests. I was thinking of collecting the results via `mbedtls_test.py` for on-target tests.